compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ResolutionEngine.java

```diff
@@ -195,13 +196,38 @@ private void mergeGraph(PackageDescriptor rootNode,
                             PackageDependencyGraphBuilder graphBuilder) {
         Collection<PackageDescriptor> directDependencies = dependencyGraph.getDirectDependencies(rootNode);
         for (PackageDescriptor directDep : directDependencies) {
+            DependencyGraph<PackageDescriptor> dependencyGraphFinal;
+            if (directDep.isBuiltInPackage()) {
```

Comment:
Do we need to change the resolution engine can we enforce this from package resolver

Reply:
AFAIU, if we are to handle this in the package resolver, we would need to create the dependency graph there. Since the dependency graph is constructed in ResolutionEngine, we have to do it here.

Reply:
This solution is good as we discussed offline.